### ISSUE TITLE
Importing ytree fails when using yt-4.0 because of yt.extern.six

### ISSUE THREAD

**Issue by virajpandya** [ytree-project/ytree#44]
# Bug report

**Bug summary**

"import ytree" fails with ModuleNotFoundError: No module named 'yt.extern.six' when using the development yt-4.0 version. 

Nate Goldbaum on slack suggested a possible solution: "ytree could probably either bundle its own copy of six or depend on six (e.g. by adding it to install_requires) in ytree's setup.py". 

**Code for reproduction**

Just  do a fresh install of yt-4.0 in a new Python3 conda environment following Cameron Hummels' instructions at https://nbviewer.jupyter.org/url/trident-project.org/notebooks/trident_demesh_install.ipynb

Then install either the stable or development version of ytree.

Then do

```python
import yt
import ytree
```

**Actual outcome**

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-19-c15a0bd11772> in <module>
      6 import yt
      7 from glob import glob
----> 8 import ytree

~/ytree/ytree/__init__.py in <module>
     14 #-----------------------------------------------------------------------------
     15 
---> 16 from ytree.data_structures import \
     17     load
     18 from ytree.data_structures.tree_node_selector import \

~/ytree/ytree/data_structures/__init__.py in <module>
     14 #-----------------------------------------------------------------------------
     15 
---> 16 from ytree.data_structures.arbor import \
     17     Arbor, \
     18     load

~/ytree/ytree/data_structures/arbor.py in <module>
     21 import os
     22 
---> 23 from yt.extern.six import \
     24     add_metaclass, \
     25     string_types

ModuleNotFoundError: No module named 'yt.extern.six'
```

**Comment by brittonsmith** [comment]
Hi @virajpandya, thanks for creating the issue for this. Unfortunately, it doesn't seem to be as simple as either removing the `extern.six` imports or adding that code to ytree. The yt-4.0 branch has a number of other changes, mostly related to using the `unyt` packages instead of `yt.units` which are also incompatible with `ytree`.

It won't be impossible to make ytree work with yt-4.0, but it will require a separate branch of ytree specifically for yt-4.0, at least until yt-4.0 is merged with its own `master` branch. This will take me a few days to get to.

**Comment by brittonsmith** [comment]
One more update. It seems that the yt-4.0 branch will be merged into yt's master branch rather soon. Once that happens, I will take care of the rest of the issues that prevent ytree from being imported with yt-4.0. Since this is so close at hand, I'd rather do this than maintain a second development branch.

**Comment by brittonsmith** [comment]
Resolved in PR #54.
